hugr/src/hugr/rewrite/outline_cfg.rs

```diff
@@ -280,16 +280,16 @@ mod test {
         h.validate(&PRELUDE_REGISTRY).unwrap();
         let backup = h.clone();
         let r = h.apply_rewrite(OutlineCfg::new([merge, tail]));
-        assert_matches!(r, Err(OutlineCfgError::MultipleExitEdges(_, _)));
+        assert!(matches!(r, Err(OutlineCfgError::MultipleExitEdges(_, _))));
         assert_eq!(h, backup);
 
         let [left, right]: [Node; 2] = h.output_neighbours(head).collect_vec().try_into().unwrap();
         let r = h.apply_rewrite(OutlineCfg::new([left, right, head]));
-        assert_matches!(r, Err(OutlineCfgError::MultipleExitNodes(a,b)) => HashSet::from([a,b]) == HashSet::from_iter([left, right, head]));
+        assert_matches!(r, Err(OutlineCfgError::MultipleExitNodes(a,b)) => assert_eq!(HashSet::from([a,b]), HashSet::from_iter([left, right])));
```

Comment:
although there were several other cases where the bool was being computed and dropped, this was the only one where the bool was false...